lib/gems/pending/openstack/openstack_handle/metric_delegate.rb

```diff
@@ -17,6 +17,7 @@ def initialize(dobj, os_handle, name)
     def list_meters(filters)
       data = []
       filters.each do |filter|
+        filter['field'] = 'resource_id' if filter['field'] == 'metadata.resource_id'
         next if filter['field'] == 'metadata.instance_id'
         raise "Unexpected Metric filter \"#{filter['field']}\"" unless filter['field'] == 'resource_id'
```

Comment:
I wish I had a better answer, but yes, putting it into _unless_ check should have same result. I understand the extra line as an another operation than in _unless_ check, so it looked more readable to me. I can change it if needed.